core/src/plugins/filed/python/mariabackup/bareos-fd-mariabackup.py

```diff
@@ -137,10 +145,81 @@ def check_plugin_options(self, mandatory_options=None):
         else:
             return bRC_OK
 
+    def check_plugin_mariadb_version(self):
+        get_version_command = self.mysqlcmd.split()
+        get_version_command.extend(
+            ["--batch", "--skip-column-names", "--execute", "select version();"]
+        )
+        DebugMessage(100, f'mariadb version check: "{get_version_command}"\n')
+        try:
+            ret = check_output(get_version_command).decode().split(".")[0]
+            mariadb_major_version = int(ret)
+            if mariadb_major_version is not None and mariadb_major_version != 0:
+                if mariadb_major_version < 11:
+                    self.checkpoints_filename = "xtrabackup_checkpoints"
+                elif mariadb_major_version >= 11:
+                    self.checkpoints_filename = "mariadb_backup_checkpoints"
+                DebugMessage(
+                    100,
+                    (
+                        f"Detected mariadb version {mariadb_major_version} "
+                        f": {self.checkpoints_filename}\n"
+                    ),
+                )
+                return bRC_OK
+        except CalledProcessError as run_err:
+            DebugMessage(
+                100,
+                f"No mariadb version detected: {run_err}\n",
+            )
+            return bRC_Error
+        except ValueError as val_err:
+            DebugMessage(
+                100,
+                f"mariadb version detected seems wrong: {val_err}\n",
+            )
+            return bRC_Error
+
+    def get_lsn_by_command(self):
+        get_lsn_cmd = self.mysqlcmd.split()
+        get_lsn_cmd.extend(
+            [
+                "--batch",
+                "--skip-column-names",
+                "--execute",
+                "SHOW ENGINE INNODB STATUS;",
+            ]
+        )
+        DebugMessage(100, f'get_lsn_by_command: "{get_lsn_cmd}"\n')
+        try:
+            ret = check_output(get_lsn_cmd).decode()
+            return ret
+        except CalledProcessError as run_err:
+            DebugMessage(
+                100,
+                f"Error while looking for LSN: {run_err}\n",
+            )
```

Comment:
```suggestion
            JobMessage(
                M_FATAL,
                f"Error while looking for LSN: {run_err}\n",
            )
```
I may be wrong, but I think the job should fail if this happens.